src/main/java/com/github/manolo8/darkbot/core/objects/facades/InventoryProxy.java

```diff
@@ -0,0 +1,60 @@
+package com.github.manolo8.darkbot.core.objects.facades;
+
+import com.github.manolo8.darkbot.Main;
+import com.github.manolo8.darkbot.core.itf.Updatable;
+import com.github.manolo8.darkbot.core.objects.swf.ObjArray;
+import eu.darkbot.api.managers.InventoryAPI;
+import eu.darkbot.util.Timer;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.ToString;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.github.manolo8.darkbot.Main.API;
+
+@Getter
+public class InventoryProxy extends Updatable implements InventoryAPI {
+    @Getter(AccessLevel.NONE)
+    private final ObjArray itemsArr = ObjArray.ofVector(true);
+
+    public List<Item> items = new ArrayList<>();
+    private final Timer updateTimer = Timer.get(10_000);
```

Comment:
maybe instead of just a hard-coded timer to re-fetch all items every 10s, also consider doing it only if some plugins is polling the api (ie: calling getItems()). Basically define getItems and make that method be the one that calls update (if timer.tryactivate) and then returns the result.